python/sdk/prelude_sdk/controllers/iam_controller.py

```diff
@@ -59,205 +35,230 @@ def update_account(self, mode: Mode = None, company: str = None, slug: str = Non
-            cfg[self.account.profile]["account"] = res_json["account"]
-            cfg[self.account.profile]["token"] = res_json["token"]
-            self.account.write_keychain_config(cfg)
-            return res_json
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
     def audit_logs(self, days: int = 7, limit: int = 1000):
         """Get audit logs from the last X days"""
         params = dict(days=days, limit=limit)
-        res = self._session.get(
+        res = self.get(
             f"{self.account.hq}/iam/audit",
             headers=self.account.headers,
             params=params,
             timeout=30,
         )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
     def subscribe(self, event: AuditEvent):
         """Subscribe to email notifications for an event"""
-        res = self._session.post(
+        res = self.post(
             f"{self.account.hq}/iam/subscribe/{event.name}",
             headers=self.account.headers,
             timeout=10,
         )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+        return res.json()
 
     @verify_credentials
     def unsubscribe(self, event: AuditEvent):
         """Unsubscribe to email notifications for an event"""
-        res = self._session.delete(
+        res = self.delete(
             f"{self.account.hq}/iam/subscribe/{event.name}",
             headers=self.account.headers,
             timeout=10,
         )
-        if res.status_code == 200:
-            return res.json()
-        raise Exception(res.text)
+        return res.json()
+
+    def sign_up(self, company, email, name, profile=None):
+        """(NOT AVAIABLE IN PRODUCTION) Create a new user and account"""
+        body = dict(company=company, email=email, name=name)
 
-    @verify_credentials
-    def accept_terms(self, name, version):
-        """Accept terms and conditions"""
         res = self._session.post(
-            f"{self.account.hq}/iam/terms",
+            f"{self.account.hq}/iam/new_user_and_account",
+            headers=self.account.headers,
+            json=body,
+            timeout=20,
+        )
+        data = res.json()
+        if profile:
+            self.account.keychain.configure_keychain(
+                data["account_id"], data["user_id"], self.account.hq, profile
+            )
+        return data
+
+
+class IAMUserController(HttpController):
+
+    def __init__(self, account):
+        super().__init__(account)
+
+    @verify_credentials
+    def list_accounts(self):
+        """List all accounts for your user"""
+        res = self.get(
+            f"{self.account.hq}/iam/user/account",
+            headers=self.account.headers,
+            timeout=10,
+        )
+        return res.json()
+
+    @verify_credentials
+    def admin_reset_password(self, email: str):
```

Comment:
i'd put this in IamAccountController, since you need to be the admin of the account in order to reset someones password